CONTRIBUTING.md

```diff
@@ -78,6 +78,18 @@ move forward.
   The AsyncAPI spec, despite describing technical behavior, is intended to be
   read by people. Use natural tone and include motivation and examples.
 
+## Breaking Change vs Non Breaking Change
+
+Specifications are not much different than software. Some changes provided in the spec can cause breaking changes for tools that support it. For example, if one of the properties that was `required` becomes `optional`, it is considered a breaking change because some tools might depend on that property and fail if all suddenly it is not provided. 
+
+_Non-breaking changes:_
+- Adding a property
+
+_Breaking changes:_
+- Making a property required
+- Making a property optional
+- Removing a property
+- Changing the type of a property in any way
```

Comment:
IMHO this sentence is not precise. I'm suggesting following one:

```suggestion
- Changing the type of a property in backward compatible way
```
**Justification:**: There are multiple ways the field type can change. AsyncAPI 2.x.y is using **UNION** structural typing, thus there are two situations that are similar, but in context of backward compatibility, different.

**Case 1 - backward compatible**

Before Type: `Server Variable Object`
After Type: `Server Variable Object | Reference Object`

**Case 2 - backward incompatible**

Before Type: `Server Variable Object | Reference Object`
After Type: `Server Variable Object`

There is also another case which is variant of `Case 1` and backward compatible as long as `Server Variable Object` and `Difference Object` have the same shape (in context of structural typing). This case can be simplified by calling it a **type renaming**.

**Case 1 (variant 1) - backward compatible**

Before Type: `Server Variable Object`
After Type: `Different Object`

Reply:
Then your suggestion should say "backward incompatible" way, right? 🤔

Reply:
@fmvilas right ;] Thanks, updated the suggestion.